runtimes/common/src/assets.rs

```diff
@@ -0,0 +1,110 @@
+// KILT Blockchain – https://botlabs.org
+// Copyright (C) 2019-2022 BOTLabs GmbH
+
+// The KILT Blockchain is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// The KILT Blockchain is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+// If you feel like getting in touch with us, you can do so at info@botlabs.org
+
+use codec::{Decode, Encode, MaxEncodedLen};
+use scale_info::TypeInfo;
+use sp_runtime::RuntimeDebug;
+use sp_std::marker::PhantomData;
+
+use kilt_asset_dids::AssetDid as AssetIdentifier;
+use public_credentials::{Config, Error, InputSubjectIdOf};
+
+#[cfg(feature = "runtime-benchmarks")]
+pub use benchmarks::*;
+
+/// Thin wrapper around the `AssetDid` type, that transform the parsing errors
+/// to errors for the public-credentials crate.
+#[derive(Clone, Eq, PartialEq, Ord, PartialOrd, RuntimeDebug, Encode, Decode, MaxEncodedLen, TypeInfo)]
+#[scale_info(skip_type_params(T))]
+#[codec(mel_bound())]
+pub struct AssetDid<T>(AssetIdentifier, Option<PhantomData<T>>);
```

Comment:
Most ideally, we would not want to plug in `T` here to use lazily `T::Config`. However, I assume you would have to touch many files to not do it this way. So we can handle this in a separate PR. I thought we already have a ticket, couldn't find it.

Reply:
Yeah, but I think this case is different. We already have a generic asset DID crate, and the public credential pallets itself does not have any knowledge of anything DID. Usually we do this in the pallet itself, which I agree is not best practice.
This is a runtime definition, so I think it is fine to have it in here, also because at some points the generic will have to be specified.

Reply:
After going deeper into the RPC implementation, I found out that any references to the runtimes should be avoided for those types that have to be used on the client side, so I changed the type a bit to remove the dependency on `T`. Please take another look when you can, as I have anyway added all the RPC part which deserves a check.